thirdeye-ui/src/app/components/anomalies-create/create-anomaly-wizard/create-anomaly-wizard.component.tsx

```diff
@@ -354,28 +354,114 @@ export const CreateAnomalyWizard: FunctionComponent<CreateAnomalyWizardProps> =
+            return {
+                start: timestamp[0],
+                end: timestamp[timestamp.length - 1],
+            };
+        }, [evaluation]);
+
+        /** @function Ensure that the date range values are always valid and make sense
+         *  @return *boolean* indicating if the attempted correction was successful or not
+         */
+        const setValidAnomalyDateRange = ([startTimeArg, endTimeArg]: [
+            number,
+            number
+        ]): boolean => {
+            if (!evaluation || !chartDataBounds) {
+                notify(
+                    NotificationTypeV1.Error,
+                    t(
+                        "message.unable-to-parse-date-range-from-the-chart-please-try-again"
+                    )
+                );
 
-            // If the list does not have any value greater than thresholdValue, return null
-            if (!nextClosestTimestamp) {
-                return null;
+                return false;
+            }
+
+            const detectionEvaluation =
+                extractDetectionEvaluation(evaluation)[0];
+            const { timestamp } = detectionEvaluation.data;
+
+            if (timestamp.length === 0) {
+                notify(
+                    NotificationTypeV1.Error,
+                    t(
+                        "message.unable-to-parse-date-range-from-the-chart-please-try-again"
+                    )
+                );
+
+                return false;
             }
 
-            return nextClosestTimestamp;
+            let start = startTimeArg;
+            let end = endTimeArg;
+
+            // Something went wrong in invoking this function since this selection is clearly wrong
+            if (end <= start) {
+                notify(
+                    NotificationTypeV1.Error,
+                    t(
+                        "message.unable-to-parse-date-range-from-the-chart-please-try-again"
+                    )
+                );
+
+                return false;
+            }
+
+            // Respect the data's Lower Bound
+            if (start < chartDataBounds.start) {
+                start = chartDataBounds.start;
+            }
+
+            // Respect the data's Upper Bound
+            if (end > chartDataBounds.end) {
+                end = chartDataBounds.end;
+            }
+
+            // Try the closest valid data-point, or fallback to the old value
+            start = timestamp.find((t) => t >= startTimeArg) || start;
+            end = timestamp.find((t) => t >= endTimeArg) || end;
+
+            // The anomaly must have a non-zero date range
+            if (start === end) {
+                // If the start AND end lie on the last point,
+                // then move the end ahead by the alert granularity
+                if (end === chartDataBounds.end) {
+                    const alertGranularityString =
+                        alertInsight?.templateWithProperties.metadata
+                            .granularity;
```

Comment:
If alertInsight is missing, you can use https://github.com/startreedata/thirdeye/blob/3d194460e662cc7ac0d88d1f9a9d381efe2693cb/thirdeye-ui/src/app/utils/visualization/visualization.util.ts#L149

Reply:
I wasn't aware of this. Thanks for the point-out, updated.